packages/autocannon-ui-desktop/main.js

```diff
@@ -0,0 +1,73 @@
+'use strict'
+
+const path = require('path')
+
+const { app, BrowserWindow } = require('electron')
+const Fastify = require('fastify')
+const { setContentSecurityPolicy } = require('electron-util')
+const Store = require('electron-store')
+const autocannonUiBackend = require('autocannon-ui-backend')
+const pkgDir = require('pkg-dir')
+
+const store = new Store()
+
+async function start() {
+  const address = await startServer()
+  await app.whenReady()
+  await createWindow(address)
+}
+
+async function startServer() {
+  const fastify = Fastify()
+  fastify.register(autocannonUiBackend)
+
+  require.resolve('autocannon-ui-frontend')
+
+  const uiRoot = path.join(
+    pkgDir.sync(require.resolve('autocannon-ui-frontend')),
+    'dist'
+  )
+  fastify.register(require('fastify-static'), {
+    root: uiRoot
+  })
+
+  const address = await fastify.listen()
+  await fastify.ready()
+  return address
+}
+
+async function createWindow(address) {
+  const win = new BrowserWindow({
+    width: 800,
+    height: 600,
+    autoHideMenuBar: true,
+    webPreferences: {
+      enableRemoteModule: false,
+      contextIsolation: true,
+      preload: path.join(app.getAppPath(), 'preload.js'),
+      additionalArguments: [JSON.stringify(store.store)]
+    }
+  })
+
+  await win.loadURL(address)
+
+  win.on('close', async () => {
+    const localStorage = await win.webContents.executeJavaScript(
```

Comment:
is this needed? what for?

Reply:
currently not, but in the future it will be helpful for saving config in local storage

Reply:
okay fair enough